plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonqFeatureDev/messages/FeatureDevMessagePublisherExtensions.kt

```diff
@@ -191,11 +192,22 @@ suspend fun MessagePublisher.sendCodeResult(
     uploadId: String,
     filePaths: List<NewFileZipInfo>,
     deletedFiles: List<DeletedFileInfo>,
-    references: List<CodeReference>
+    references: List<CodeReferenceGenerated>
 ) {
-    // It is being done this mapping as featureDev currently doesn't support fully references.
-    val refs = references.filter { it.licenseName != null && it.url != null && it.repository != null }.map {
-        ReducedCodeReference(information = it.licenseText())
+    val refs = mutableListOf<CodeReference>()
+    if (references.isNotEmpty()) {
+        refs += references.map { ref ->
+            CodeReference(
+                licenseName = ref.licenseName,
+                repository = ref.repository,
+                url = ref.url,
+                recommendationContentSpan = RecommendationContentSpan(
+                    ref.recommendationContentSpan?.start ?: 0,
+                    ref.recommendationContentSpan?.end ?: 0,
+                ),
+                information = "Reference code under **${ref.licenseName}** license from repository [${ref.repository}](${ref.url})"
+            )
+        }
     }
```

Comment:
```suggestion
    val refs = references.map { ref ->
        CodeReference(
            licenseName = ref.licenseName,
            repository = ref.repository,
            url = ref.url,
            recommendationContentSpan = RecommendationContentSpan(
                ref.recommendationContentSpan?.start ?: 0,
                ref.recommendationContentSpan?.end ?: 0,
            ),
            information = "Reference code under **${ref.licenseName}** license from repository [${ref.repository}](${ref.url})"
        )
    }
```

Reply:
We can have `references` returned as emptyList as well from our backend, so we have added the check for if it not empty.

Reply:
in another thread, discussed how `map{}` automatically deals with the empty case